### DISCUSSION TITLE
Download news every X: hour to the Azuracast.

### DISCUSSION THREAD

**Discussion by RadioHerzblut** [AzuraCast/AzuraCast#6722] [General]
Hello.

I have installed the following version on Ubuntu:
v0.19.1 Stable - Docker - PHP 8.2

I am looking for an Azuracast a way to download an MP3 file always 
at X:50 minutes and put it into the playlist at X:00. 

Currently you can set a start and end time from Monday to Saturday. Also the date from wa'nn to when. 

But I need this file every hour in the playlist ( news ).

Does anyone have an idea?

**Comment by Moonbase59** [comment]
I’d go the pragmatic way:

- Mount a host folder into your station media, say `/var/azuracast/news`.
- Write a script that periodically fetches the news, use crontab for executing it.
- Let it always write to the same file, say `/var/azuracast/news/hourly_news.mp3`. Use atomic moves.
- Then you can simply include `hourly_news.mp3` in your normal playlist/rotation rules. If done correctly, it’ll always present the current news. (Content changes, name stays.)

This is the same scheme I use with my [TOTH announcements](https://github.com/AzuraCast/AzuraCast/discussions/6294), which uses a time announcement file that gets updated every minute and thus can always be used.

**Comment by RadioHerzblut** [comment]
I have no idea how to write scripts. Adjusting paths and file names is no problem. 

Do you perhaps have such a script which you would make available?

**Comment by Moonbase59** [comment]
Unfortunately I don’t have any ready-made samples. You might want to google "bash script" and make yourself familiar with tools like `curl` (`curl --help` or `man curl`) and `wget` (`wget --help` or `man wget`). It’s really easy.

A super-simple bash script for copying a news mp3 could look as easy as this:
```bash
#!/bin/bash
curl -s -o /var/azuracast/news/hourly_news.mp3 https://your.newsprovider.com/path/to/news.mp3
```

Assuming you saved this on your AzuraCast server as `/usr/local/bin/get-news`, you could automatically run it every hour at xx:50 using `crontab -e` to edit your crontab file, and add a line like this:
```crontab
# fetch the news every hour at xx:50
50 * * * * /usr/local/bin/get-news
```

**Comment by RadioHerzblut** [comment]
That's one approach. Thanks for the information and signposts ;-)

**Reply by RadioHerzblut** [reply]
With Cronjob an wget work it´s finde ;-)
